exo/tinychat/index.js

```diff
@@ -72,6 +72,28 @@ document.addEventListener("alpine:init", () => {
       return `${s}s`;
     },
 
+    async populateSelector() {
+      const response = await fetch(`${this.endpoint}/modelpool`);
```

Comment:
the code above shows:

```endpoint: `${window.location.origin}/v1`,```

so either you need to change the backend to use /v1/modelpool, or change this to `const response = await fetch(`${window.location.origin}/modelpool`);`

Reply:
I think that this function overall wants to look something like this:

```
async populateSelector() {
      try {
        const response = await fetch(`${window.location.origin}/modelpool`);
        if (!response.ok) {
          const errorResBody = await response.json();
          throw new Error(`Failed to get model pool: ${errorResBody?.detail || 'Unknown error'}`);
        }

        const responseJson = await response.json();
        const sel = document.querySelector(".model-select");
        if (!sel) {
          throw new Error("Could not find model selector element");
        }

        // Clear the current options and add new ones
        sel.innerHTML = '';
          
        const modelDict = responseJson["model pool"];

        Object.entries(modelDict).forEach(([key, value]) => {
          const opt = document.createElement("option");
          opt.value = key;
          opt.textContent = value;
          sel.appendChild(opt);
        });

        // Set initial value to the first model
        const firstKey = Object.keys(modelDict)[0];
        if (firstKey) {
          sel.value = firstKey;
          this.cstate.selectedModel = firstKey;
        }
      } catch (error) {
        console.error("Error populating model selector:", error);
        this.errorMessage = error.message;
      }
    },

```